util/src/test/java/io/kubernetes/client/CopyTest.java

```diff
@@ -67,14 +69,16 @@ public void testUrl() throws IOException, ApiException, InterruptedException {
                     .withStatus(404)
                     .withHeader("Content-Type", "application/json")
                     .withBody("{}")));
-
+    Semaphore s = new Semaphore(1);
+    s.acquire();
     try {
       copy.copyFileFromPod(pod, "container", "/some/path/to/file");
     } catch (IOException | ApiException e) {
       e.printStackTrace();
+    } finally {
+      s.release();
     }
-
-    Thread.sleep(2000);
+    s.acquire();
```

Comment:
in this particular unit test, the client will be starting a websocket handshake against the wiremock server but because wiremock doesn't fully support websocket so the process of hand-shaking will be aborted abnormally. out of some reason, the wiremock will take a second-ish period to count the unfinished requests.. i guess that's why we have to sleep in this test..

Reply:
I re-did how we initiate web sockets to make it more synchronous and resolve this.